eventually-v0-5-0/src/lib.rs

```diff
@@ -0,0 +1,73 @@
+#![deny(unsafe_code, unused_qualifications, trivial_casts)]
+#![deny(clippy::all)]
+#![warn(clippy::pedantic)]
+
+pub mod event;
+pub mod metadata;
+pub mod test;
+pub mod version;
+
+use serde::{Deserialize, Serialize};
+
+use crate::metadata::Metadata;
+
+pub type Messages<T> = Vec<Message<T>>;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Message<T> {
+    pub payload: T,
+    pub metadata: Metadata,
+}
+
+impl<T> Message<T> {
+    pub fn with_metadata<F>(mut self, f: F) -> Self
```

Comment:
why use a factory function to generate the metadata, rather than leaving that to the calling code? how would this be used?

Reply:
This doesn't generate metadata, it's a method to enrich the metadata the event currently has.

I'm thinking of using this function in some decorator types, to set specific metadata like timestamps or other useful information for debugging.

Reply:
i think the trait bound would at least have to be `FnMut` in that case.

do you have a code snippet to show what you mean?

Reply:
https://github.com/get-eventually/eventually-rs/blob/d8a3b1e67ab332d3451c47c5b387b9b58f985b46/eventually-v0-5-0/src/lib.rs#L62-L66

You don't need `FnMut` because the function is taking ownership of `Metadata` and returning a new value.